storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -128,56 +308,121 @@ private void updateSupervisorsResources(Cluster cluster, Topologies topologies)
-            } else {
-                str.append("/** Node: Unknown may be dead -" + entry.getKey() + " **/\n");
+    private void initialize(Topologies topologies, Cluster cluster) {
+        this.cluster = cluster;
+        this.topologies = topologies;
+        this.nodes = new RAS_Nodes(this.cluster, this.topologies);
+        initUsers(topologies, cluster);
+    }
+
+    /**
+     * Get resource guarantee configs
+     *
+     * @return
+     */
+    private Map<String, Map<String, Double>> getUserResourcePools() {
+        Object raw = this.conf.get(Config.RESOURCE_AWARE_SCHEDULER_USER_POOLS);
+        Map<String, Map<String, Double>> ret = new HashMap<String, Map<String, Double>>();
+
+        if (raw != null) {
+            for (Map.Entry<String, Map<String, Number>> UserPoolEntry : ((Map<String, Map<String, Number>>) raw).entrySet()) {
+                String user = UserPoolEntry.getKey();
+                ret.put(user, new HashMap<String, Double>());
+                for (Map.Entry<String, Number> resourceEntry : UserPoolEntry.getValue().entrySet()) {
+                    ret.get(user).put(resourceEntry.getKey(), resourceEntry.getValue().doubleValue());
+                }
             }
-            for (Map.Entry<String, Map<WorkerSlot, Collection<ExecutorDetails>>> topo_sched : schedulingMap.get(entry.getKey()).entrySet()) {
-                str.append("\t-->Topology: " + topo_sched.getKey() + "\n");
-                for (Map.Entry<WorkerSlot, Collection<ExecutorDetails>> ws : topo_sched.getValue().entrySet()) {
-                    str.append("\t\t->Slot [" + ws.getKey().getPort() + "] -> " + ws.getValue() + "\n");
+        }
+
+        Map fromFile = Utils.findAndReadConfigFile("user-resource-pools.yaml", false);
+        Map<String, Map<String, Number>> tmp = (Map<String, Map<String, Number>>) fromFile.get(Config.RESOURCE_AWARE_SCHEDULER_USER_POOLS);
+        if (tmp != null) {
+            for (Map.Entry<String, Map<String, Number>> UserPoolEntry : tmp.entrySet()) {
+                String user = UserPoolEntry.getKey();
+                ret.put(user, new HashMap<String, Double>());
+                for (Map.Entry<String, Number> resourceEntry : UserPoolEntry.getValue().entrySet()) {
+                    ret.get(user).put(resourceEntry.getKey(), resourceEntry.getValue().doubleValue());
                 }
             }
         }
-        return str.toString();
+        return ret;
+    }
+
+    private SchedulingState checkpointSchedulingState() {
+        LOG.debug("/*********Checkpoint scheduling state************/");
+        for (User user : this.getUserMap().values()) {
+            LOG.debug(user.getDetailedInfo());
+        }
+        LOG.debug(ResourceUtils.printScheduling(this.cluster, this.topologies));
+        LOG.debug("nodes:\n{}", this.nodes);
+        LOG.debug("/*********End************/");
+        return new SchedulingState(this.userMap, this.cluster, this.topologies, this.nodes, this.conf);
+    }
```

Comment:
Couldn't we log these things in the constructor for SchedulingState?  If so, we can remove this method.